cmds/rm/rm.go

```diff
@@ -37,27 +34,20 @@ func init() {
 		os.Args[0] = cmd
 		defUsage()
 	}
-	flag.BoolVar(&flags.i, "i", false, "Interactive mode.")
-	flag.BoolVar(&flags.v, "v", false, "Verbose mode.")
-	flag.BoolVar(&flags.r, "R", false, "Remove file hierarchies")
-	flag.BoolVar(&flags.r, "r", false, "Equivalent to -R.")
-	flag.Parse()
 }
 
 func rm(files []string) error {
 	f := os.Remove
-	if flags.r {
+	if *hierarchies {
 		f = os.RemoveAll
```

Comment:
When you do

```sh
rm -ri /foo/bar
```

where 
```
/foo/bar
/foo/bar/something
/foo/bar/somethingelse
```

Shouldn't it ask about all files separately? The way this is written, it'll only ask about `/foo/bar`, but not `/foo/bar/something`, or `/foo/bar/somethingelse`.